plugins/BEdita/API/src/Utility/JsonApi.php

```diff
@@ -95,7 +97,14 @@ protected static function formatItem($item, $type = null)
         }
         unset($attribute);
 
-        return compact('id', 'type', 'attributes');
+        if (!$showLink) {
+            return compact('id', 'type', 'attributes');
+        }
+
+        $links = [];
+        $links['self'] = Router::fullBaseUrl() . '/' . $type . '/' . $id;
```

Comment:
I'd avoid building URLs this way… We might use [named routes](http://book.cakephp.org/3.0/en/development/routing.html#using-named-routes), but this could be a future improvement: this works fine for now. :smile: